core/trino-main/src/main/java/io/trino/SystemSessionProperties.java

```diff
@@ -326,6 +326,13 @@ public SystemSessionProperties(
                         QUERY_MAX_RUN_TIME,
                         "Maximum run time of a query (includes the queueing time)",
                         queryManagerConfig.getQueryMaxRunTime(),
+                        value -> {
+                            queryManagerConfig.getQueryMaxRunTimeHardLimit().ifPresent(hardLimit -> {
+                                if (value.compareTo(hardLimit) > 0) {
+                                    throw new TrinoException(INVALID_SESSION_PROPERTY, format("%s must not exceed hard limit %s: %s", QUERY_MAX_RUN_TIME, hardLimit, value));
```

Comment:
It's a good idea to mention the source of the limit (i.e. name of config property which set the hard limit) in error message. WDYT @findepi ?

Reply:
Changed to `query_max_run_time must not exceed 'query.max-run-time.hard-limit' %s: %s`, a little repetitive though 🙄